lib/rubocop/cop/style/no_metaprogramming.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Discourages the use of metaprogramming. It's designed
+      # to be enabled in select folders within your application. It was
+      # originally developed with the idea of "Business logic should minimize
+      # the amount of metaprogramming."
+      #
+      # This cop specifically will warn on usage of `.included`, `.inherited`,
+      # `#method_missing`, `.define_method`, `#instance_eval`, `.class_eval`,
+      # and `.define_singleton_method`.
+      #
+      class NoMetaprogramming < Base
+        def_node_matcher :using_method_missing?, <<~PATTERN
+          (def :method_missing ...)
+        PATTERN
+
+        METHOD_MISSING_MSG = 'Please do not use method_missing. Instead, explicitly define the'\
```

Comment:
I'd be nice if all your messages followed this format, as I see in some of them it's inverted. Let's avoid this inconsistency. Also - please put backticks around all method names referred in those messages. (e.g. `method_missing`).